### DISCUSSION TITLE
Create Release/Upload Release Assets is Successful, but no Webhook?

### DISCUSSION THREAD

**Discussion by szul** [community/community#26454] [Actions]
<p>I have an action that successful creates a release and then uploads release assets to it after the release creation; however, I’ve noticed that the webhook triggers for <code>created</code>, <code>released</code>, and <code>published</code> are triggered only for release creation, and no webhook is triggered for when the release is modified with the assets. This results in my webhook looking for assets that aren’t there yet.</p>
<p>Is there a preferred method for accomplishing this? Is there no webhook for when assets are uploaded?</p>

**Comment by szul** [comment]
<p>Look into this a possible solution:</p>
<p><a href="https://github.com/marketplace/actions/github-release-create-update-and-upload-assets" rel="nofollow noopener">https://github.com/marketplace/actions/github-release-create-update-and-upload-assets</a></p>

**Comment by szul** [comment]
<p>Maybe not. Ran into:</p>
<p><a href="https://github.com/meeDamian/github-release/issues/9" rel="nofollow noopener">https://github.com/meeDamian/github-release/issues/9</a></p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/szul">@szul</a> ,</p>
<p>Did you use the <strong>GITHUB_TOKEN</strong> to authenticate on the action for release?</p>
<p>There is a note you need to know, you also can reference the docs about <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#using-the-github_token-in-a-workflow" rel="nofollow noopener"><strong>authenticating with the GITHUB_TOKEN</strong></a>:</p>
<blockquote>
<p>Events triggered by the GITHUB_TOKEN will not create a new workflow run, this prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>
<p>If you want the release event generated by the action can trigger the workflow runs, you can create a personal access token and set it as a secret in your repository, then use this personal access token to authenticate. But you also should avoid the recursive workflow runs.</p>
<p>If you have used a personal access token to authenticate but the workflow runs still are not triggered? Please share your repository and the workflow YAML file with us so that we can check more detailed configurations your workflow to analyze the root cause.</p>

**Comment by szul** [comment]
<p>The release works fine. The problem is its a two step process:</p>
<p>* Create the “release”</p>
<p>* Add assets to the release</p>
<p>The first part (release creation) seems to generate the <code>created</code>, <code>released</code>, <code>published</code> actions for the webhooks. There doesn’t seem to be any action for updating a release with assets, so when step 2 above occurs, the webhook is not pinged.</p>
<p>The two GitHub-branded Actions for this are <code>create-release</code> and <code>upload-release-asset</code>, which is why it’s a two-step process.</p>
<p>I’m working on building my own Action to see if it works via changing draft status.</p>

**Comment by szul** [comment, marked as answer]
<p>I resolved my problem by forking the above repo and creating a new action from it:</p>
<p><a href="https://github.com/UVASOMIT/action-release-with-assets" rel="nofollow noopener">https://github.com/UVASOMIT/action-release-with-assets</a></p>
<p>With this action you can specify the release and the assets in one step. The release is initially created as a draft, and once the assets are added, it is updated to remove the draft status. When you check your webhooks, <code>release</code> and <code>published</code> both contain the assets.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/szul">@szul</a> ,</p>
<p>Glad that you have solved this question.</p>
<p>Yeah, currently we there is not any activity type of the release event can trigger workflow when uploading assets to a created release. The activity <strong>release.edited</strong> seems does not include uploading/deleting assets from the created release.</p>
<p>I think your solution should be the best way currently, and thanks for sharing your solution.</p>
